backend/GymBeam/CommandHandlers/AuthenticationCommandHandler.cs

```diff
@@ -0,0 +1,48 @@
+﻿using Domain;
+using Domain.Exceptions;
+using GymBeam.Commands;
+using GymBeam.Response;
+using GymBeam.Utils;
+using MediatR;
+
+namespace GymBeam.CommandHandlers
+{
+    public class AuthenticationCommandHandler
+        : IRequestHandler<RegisterCommand, UserResponse>
+    {
+        private readonly IRepository _repository;
+
+        public AuthenticationCommandHandler(IRepository repository)
+        {
+            _repository = repository;
+        }
+
+        Task<UserResponse> IRequestHandler<RegisterCommand, UserResponse>.Handle(RegisterCommand request, CancellationToken cancellationToken)
+        {
+            var userExists = _repository.GetUsers(x => x.Name == request.Username).FirstOrDefault();
+            if (userExists != null)
+                throw new UserAlreadyExistsException(request.Username);
+
+            var hash = ShaHelper.QuickHash(request.Password);
+            var user = new User 
+            { 
+                Name = request.Username,
+                DisplayName = request.DisplayName,
+                ReservationDisabled = false,
+                HashedPassword = hash,
+                Role = Role.User
+            };
+            _repository.Add(user);
+            _repository.SaveChangesAsync();
+
+            return Task.FromResult(new UserResponse
+            {
+                Id = user.Id,
+                Name = user.Name,
+                DisplayName = user.DisplayName,
+                Role = user.Role.ToString(),
+                ReservationDisabled = user.ReservationDisabled,
+            });
```

Comment:
Do we really need to return UserResponse here? As long as we don't plan to automatically login the user after registration, this data won't be used for anything, so returning it to the user is unnecessary. 
Also: Is returning business data as a result of CommandHandler recommended in CQRS?